client/src/pages/index.tsx

```diff
@@ -207,7 +217,7 @@ const Homepage: NextPage<FeedsPageInitials> = (props): JSX.Element => {
 			<div style={{ marginTop: -10 }}>
 				{trendingTags.map(tag => {
 					return (
-						<div key={tag._id} onClick={() => setTagFilter(`${tagFilter},${tag.name}`)} className={'custom-tag'} style={{ backgroundColor: stringToColor(tag.name), cursor: 'pointer' }}>
+						<div key={tag._id} onClick={() => handleFilteringTags(tag.name)} className={'custom-tag'} style={{ backgroundColor: stringToColor(tag.name), cursor: 'pointer' }}>
```

Comment:
please use `useCallback` instead the callback, and the dependency should be `trendingTags `